mesonbuild/environment.py

```diff
@@ -568,7 +568,8 @@ def __init__(self, source_dir: T.Optional[str], build_dir: T.Optional[str], opti
         if bt in self.options and (db in self.options or op in self.options):
             mlog.warning('Recommend using either -Dbuildtype or -Doptimization + -Ddebug. '
                          'Using both is redundant since they override each other. '
-                         'See: https://mesonbuild.com/Builtin-options.html#build-type-options')
+                         'See: https://mesonbuild.com/Builtin-options.html#build-type-options',
+                         fatal=False)
```

Comment:
This is something users can presumably fix, as we're nagging them about the CLI options they pass, on the other hand idk if I care.

Reply:
This is one people complain about all the time, since they want to set their buildtype to something, and their optimization level to something else, like `-Dbuildtype=release -Doptimization=3` (or 2) seems like a legitimate thing to do, ie: "I'm building release, but optlevel 2 isn't fast enough or optlevel 3 breaks." Making that a hard error seems not very nice.